crates/bevy_input/src/keyboard.rs

```diff
@@ -427,6 +438,177 @@ pub enum KeyCode {
     Cut,
 }
 
+#[cfg(feature = "winit_extra")]
+impl From<winit::event::VirtualKeyCode> for KeyCode {
```

Comment:
I understand the desire to support `From`, as it is generally recommended to support both `Into` and `From` with the single `From` impl. But I'm not a fan of using this approach here. The point of `bevy_input` and `bevy_window` is to be agnostic interfaces for windowing and input plugins. The point of `bevy_winit` is to be specific implementations of those plugins.

The approach here blurs the lines (at the cost of legibility and code complexity) for _very_ dubious gains. I think these should be converted to `Into` trait impls in the `bevy_winit` crate. And we should revert all of the "optional winit dependency" stuff.

Reply:
You cannot do that in rust. It complains that one of the from or into types must be in the current crate. Since the input events are in the bevy_input crate we cannot implement From trait inside the bevy_winit crate.

Reply:
I would argue that it blurs the lines. It adds complexity on the configuration level where you enable or disable features but rust features are there for just that reason.